File: gdx-runtime/src/com/mbrlabs/mundus/runtime/Shaders.java

package com.mbrlabs.mundus.runtime;

import com.mbrlabs.mundus.commons.shaders.DepthShader;
import com.mbrlabs.mundus.commons.shaders.SkyboxShader;

public class Shaders {

    private final SkyboxShader skyboxShader;
    private final DepthShader depthShader;

    public Shaders() {
        skyboxShader = new SkyboxShader();
        skyboxShader.init();
        depthShader = new DepthShader();
        depthShader.init();
    }

    public SkyboxShader getSkyboxShader() {
        return skyboxShader;
    }

    public DepthShader getDepthShader() {
        return depthShader;
    }
}

File: editor/src/main/com/mbrlabs/mundus/editor/shader/Shaders.kt

package com.mbrlabs.mundus.editor.shader

import com.badlogic.gdx.graphics.glutils.ShaderProgram
import com.badlogic.gdx.utils.Disposable
import com.mbrlabs.mundus.commons.shaders.DepthShader
import com.mbrlabs.mundus.commons.shaders.SkyboxShader
import com.mbrlabs.mundus.editor.tools.picker.PickerShader

object Shaders : Disposable {

    val wireframeShader: WireframeShader
    val skyboxShader: SkyboxShader
    val pickerShader: PickerShader
    val depthShader: DepthShader

    init {
        ShaderProgram.pedantic = false
        wireframeShader = WireframeShader()
        wireframeShader.init()
        skyboxShader = SkyboxShader()
        skyboxShader.init()
        pickerShader = PickerShader()
        pickerShader.init()
        depthShader = DepthShader()
        depthShader.init()
    }

    override fun dispose() {
        wireframeShader.dispose()
        skyboxShader.dispose()
        pickerShader.dispose()
        depthShader.dispose()
    }

}
